models/taskRequests.js

```diff
@@ -0,0 +1,119 @@
+const { TASK_REQUEST_STATUS } = require("../constants/taskRequests");
+const { TASK_STATUS } = require("../constants/tasks");
+const firestore = require("../utils/firestore");
+const taskRequestsCollection = firestore.collection("taskRequests");
+const tasksModel = require("./tasks");
+const userModel = require("./users");
+
+/**
+ * Fetch all task requests
+ *
+ * @return {Object}
+ */
+const fetchTaskRequests = async () => {
+  const taskRequests = [];
+
+  const visitedUsersId = new Set();
+  const visitedUsers = [];
+
+  const taskRequestsSnapshot = (await taskRequestsCollection.get()).docs;
+
+  for (const taskRequestSnapshot of taskRequestsSnapshot) {
+    const taskRequest = taskRequestSnapshot.data();
+    const { taskData } = await tasksModel.fetchTask(taskRequest.taskId);
+    const users = await Promise.all(
+      taskRequest.requestors.map((requestor) => {
+        if (visitedUsers.includes(requestor)) {
+          return visitedUsersId.find((visitedUserId) => (visitedUserId.id = requestor));
+        }
+        return userModel.fetchUser({ userId: requestor });
+      })
+    );
+
+    taskRequests.push({ id: taskRequestSnapshot.id, ...taskRequest, task: taskData, requestors: users });
+  }
+
+  return taskRequests;
+};
+
+/**
+ * Creates a task request
+ *
+ * @param taskId { string }: id of task request
+ * @return {Promise<{taskRequest: Object}>}
+ */
+const addOrUpdate = async (taskId, userId) => {
+  try {
+    const taskRequestsSnapshot = await taskRequestsCollection.where("taskId", "==", taskId).get();
+    const [taskRequestRef] = taskRequestsSnapshot.docs;
```

Comment:
The bug is that the docs property of the taskRequestsSnapshot object is an array of documents, but the [0] index is being used to access a single document. This will cause an error if there are no documents in the array. To fix this bug, you need to use the find() method to find the document with the specified taskId. You can do this by adding the following line to the code:

Reply:
I get your point, but since this is javascript I would instead get a `undefined` as value instead of code breaking on this, and I am using this `undefined` result to check if the task request data exists.

Reply:
@RitikJaiswal75  same here